### ISSUE TITLE
useref only works on server mode

### ISSUE THREAD

**Issue by ai0** [mamboer/hexo-filter-cleanup#2]
When `hexo s`, the useref works pretty well,
but `hexo s -s`, the useref doesn't work, which means, the concat target file return 404 error.

And when `hexo g`, the 404 error occurred again, and in the public folder, there are no concat target file generated, but in cli it shows:

```
INFO  hexo-filter-cleanup(USEREF): css/combined.css
...
INFO  hexo-filter-cleanup(USEREF): css/combined.js
```

Relative info is as follows:

> $ hexo version
> 
> hexo: 3.2.2
> hexo-cli: 1.0.2
> os: Linux 4.4.16-moby linux x64
> http_parser: 2.7.0
> node: 6.2.0
> v8: 5.0.71.47
> uv: 1.9.1
> zlib: 1.2.8
> ares: 1.10.1-DEV
> icu: 57.1
> modules: 48
> openssl: 1.0.2h
> 
> $ npm -v
> 3.8.9
> 
> $ node -v
> v6.2.0

**Comment by aispin** [comment]
hi, 

`hexo-filter-cleanup`'s features were tested under node v5.x and it works like a charm.

Try to switch your node version to 5.x or 4.x, because the codebase of hexo isn't compatible with 6.x currently. 

If it still not works, pls send me your copy of source codes so that i can debug them deeply.

Thanks.

**Comment by ai0** [comment]
Hi,

thanks for reply,
I had tried node v5.12.0 and node v4.4.7,
either of them only works well on server mode like on v6.2.0,
and after `hexo g`, those static file also return 404 error.

I created a minimal container for you to debug it,
to reappear my situation,
if you had docker installed, use these commands:

```
git clone https://github.com/ai0/Hexo-debug
cd Hexo-debug
make run-test
```

and then there will had a environment with node 5.x,
if you want node 4.x, just edit the Dockerfile line 1 `FROM mhart/alpine-node:5` to `FROM mhart/alpine-node:4` before you run `make run-test`.

Inside the container, run `hexo s` and then visit `http://127.0.0.1:4000/css/hexo.css` or `http://127.0.0.1:4000/js/hexo.js` it will be 200 OK,
but run `hexo s -s` or  after `hexo g` the hexo.css and hexo.js will 404 error.

Thank you again for the convenient plugin.

**Comment by aispin** [comment]
![image](https://cloud.githubusercontent.com/assets/85879/17671572/4d3f1ae6-634a-11e6-8d04-87d69e8deb1e.png)
Hi, sorry for replying you so late.
I couldn't make the docker container work in my local machine, it seems that there are certain network issues.

I've reviewed the codes related to the useref and did some fixes.

Could you pls  upgrade the `hexo-filter-cleanup` plugin to the lastest version and run your test again?

Or, you can take a look at this project (https://github.com/o2team/events), it is using this plugin and it works well. Just folk and have a try.

Best regards.

**Comment by ai0** [comment]
Nice work!

I am so excited to say that after I upgraded `hexo-filter-cleanup` to v1.0.5 and then everything works great now!
The `hexo s -s` is works like a charm like you said and no 404 Error as well as `hexo -g`, the issue has been fixed now.

From the screenshot it seems like the docker engine not connect to the corresponding docker hub, or encounter some network hijack,
and the dependent container is hosted on official docker hub:

[https://hub.docker.com/r/mhart/alpine-node/](https://hub.docker.com/r/mhart/alpine-node/)

it is real existed, should works fine.

Thank you very very much for solve this issue!

Best regards.
